test/e2e/opentelemetry/opentelemetry.test.ts

```diff
@@ -43,31 +33,47 @@ createNextDescribe(
       return span
     }
     const sanitizeSpans = (spans: SavedSpan[]) => {
-      const seenSpans = new Set()
       return spans
+        .sort((a, b) =>
+          (a.attributes?.['next.span_name'] ?? '').localeCompare(
+            b.attributes?.['next.span_name'] ?? ''
+          )
+        )
         .sort((a, b) =>
           (a.attributes?.['next.span_type'] ?? '').localeCompare(
             b.attributes?.['next.span_type'] ?? ''
           )
         )
         .map(sanitizeSpan)
-        .filter((span) => {
-          const target = span.attributes?.['http.target']
-          const result =
-            !span.attributes?.['http.url']?.startsWith('http://localhost') &&
-            !seenSpans.has(target)
-
-          if (target) {
-            seenSpans.add(target)
-          }
+    }
 
-          return result
-        })
+    /**
+     * Routes can be executed multiple times per request in dev.
+     */
+    const removeDevDuplicateSpans = (spans: SavedSpan[]) => {
```

Comment:
If we want to hide the internal requests from traces we shouldn't need to remove and duplicate spans I think. 

Seems like these requests need to be skipped from tracing as well as they are also internal

https://github.com/vercel/next.js/blob/26e8197d7b3f3c2d478abf09b642a0fcc03849f6/packages/next/src/server/next-server.ts#L1433
https://github.com/vercel/next.js/blob/26e8197d7b3f3c2d478abf09b642a0fcc03849f6/packages/next/src/server/next-server.ts#L2270

Reply:
`fetch` have parentSpan they are internal.

The span that is duplicated is `BaseServer.handleRequest` which is the `root` span.

The best solution would be to hide those spans too, but I'm not sure how to do it easily.

Thanks for pointing-out those two fetches, will add them.

Reply:
I could not reproduce that fetch in tests. When is that `fetch` called?

But I've disabled it anyway.

Reply:
@ijjk removed these fetches in followup PRs :+1: